src/Nethermind/Nethermind.Synchronization/ParallelSync/SyncProgressResolver.cs

```diff
@@ -117,6 +121,25 @@ public long FindBestFullState()
                 }
             }
 
+            if (bestFullState == 0)
+            {
+                if (bestStateCounter != -1)
+                {
+                    return bestStateCounter;
+                }
+                byte[] bestStatePersistedCounter = _stateDb.Get(LatestBlockWithFullState);
+                bestStateCounter = bestStatePersistedCounter != null ? BitConverter.ToInt64(bestStatePersistedCounter) : 0;
+                return bestStateCounter;
+            }
+            else
+            {
+                if (bestFullState != bestStateCounter)
+                {
+                    _stateDb.Set(LatestBlockWithFullState, BitConverter.GetBytes(bestFullState));
```

Comment:
To consider:
- how often it's going to be called and the impact on the performance
- is it possible to be called from multiple threads

Reply:
> how often it's going to be called and the impact on the performance
It's called every second, _bestStateCounter is used to reduce 1 call to the database.
> is it possible to be called from multiple threads
In theory - yes, let me add me add proper long value setting